ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/ValidateUtilityNetworkTopology/ValidateUtilityNetworkTopology.cpp

```diff
@@ -0,0 +1,671 @@
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "ArcGISFeature.h"
+#include "ArcGISFeatureListModel.h"
+#include "ArcGISFeatureTable.h"
+#include "AttributeListModel.h"
+#include "AuthenticationChallenge.h"
+#include "CodedValueDomain.h"
+#include "Credential.h"
+#include "Envelope.h"
+#include "Error.h"
+#include "Feature.h"
+#include "FeatureEditResult.h"
+#include "FeatureLayer.h"
+#include "FeatureTable.h"
+#include "FeatureTableEditResult.h"
+#include "Field.h"
+#include "Graphic.h"
+#include "GraphicListModel.h"
+#include "GraphicsOverlay.h"
+#include "GraphicsOverlayListModel.h"
+#include "IdentifyLayerResult.h"
+#include "LabelDefinition.h"
+#include "LabelDefinitionListModel.h"
+#include "LayerListModel.h"
+#include "LoadSettings.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MapTypes.h"
+#include "PictureMarkerSymbol.h"
+#include "Point.h"
+#include "Portal.h"
+#include "PortalItem.h"
+#include "QueryParameters.h"
+#include "ServiceFeatureTable.h"
+#include "ServiceGeodatabase.h"
+#include "ServiceTypes.h"
+#include "ServiceVersionInfo.h"
+#include "ServiceVersionParameters.h"
+#include "SimpleLabelExpression.h"
+#include "SimpleMarkerSymbol.h"
+#include "SpatialReference.h"
+#include "SubtypeSublayer.h"
+#include "SymbolTypes.h"
+#include "TaskTypes.h"
+#include "TextSymbol.h"
+#include "UtilityAssetGroup.h"
+#include "UtilityAssetType.h"
+#include "UtilityDomainNetwork.h"
+#include "UtilityElement.h"
+#include "UtilityElementTraceResult.h"
+#include "UtilityNetwork.h"
+#include "UtilityNetworkCapabilities.h"
+#include "UtilityNetworkDefinition.h"
+#include "UtilityNetworkListModel.h"
+#include "UtilityNetworkSource.h"
+#include "UtilityNetworkState.h"
+#include "UtilityNetworkTypes.h"
+#include "UtilityNetworkValidationJob.h"
+#include "UtilityTerminal.h"
+#include "UtilityTerminalConfiguration.h"
+#include "UtilityTier.h"
+#include "UtilityTraceParameters.h"
+#include "UtilityTraceResult.h"
+#include "UtilityTraceResultListModel.h"
+#include "UtilityValidationResult.h"
+#include "ValidateUtilityNetworkTopology.h"
+#include "Viewpoint.h"
+
+#include <QUuid>
+#include <QFuture>
+
+using namespace Esri::ArcGISRuntime;
+
+
+ValidateUtilityNetworkTopology::ValidateUtilityNetworkTopology(QObject* parent /* = nullptr */) :
+  QObject(parent),
+  m_cred(new Credential("editor01", "S7#i2LWmYH75", this))
+{
+  m_portal = new Portal(QUrl("https://sampleserver7.arcgisonline.com/portal/sharing/rest"), m_cred, this);
+
+  m_portalItem = new PortalItem(m_portal, "6e3fc6db3d0b4e6589eb4097eb3e5b9b", this);
+
+  m_map = new Map(m_portalItem, this);
```

Comment:
It looks like most of these members aren't referenced again. I think these can be created locally here. No need for members.

Reply:
changed